src/lib/Scenes/City/City.tsx

```diff
@@ -42,31 +42,34 @@ export class CityView extends Component<Props, State> {
     { id: "museums", text: "Museums" },
   ]
 
+  handleEvent = ({
+    filter,
+    buckets,
+    cityName,
+    relay,
+    sponsoredContent,
+  }: {
+    filter: Tab
+    buckets: BucketResults
+    cityName: string
+    relay: RelayProp
+    sponsoredContent: { introText: string; artGuideUrl: string }
+  }) => {
+    this.setState({
+      buckets,
+      filter,
+      cityName,
+      relay,
+      sponsoredContent,
+    })
+  }
```

Comment:
Made this a bound method so that it will be the exact same instance when we subscribe and unsubscribe.